### PR TITLE
[POS][Custom payment UI] – Reader not connected error handling

### LINKED ISSUE #13005: [UI] Show "reader not connected" prompt in Totals
Show "reader not connected" message in Totals and "Connect to reader" button, allowing connecting to a reader from the Totals screen.

### PR DESCRIPTION
**Pull request by samiuelson**
Closes: #13005

### Description

This PR: 
1. Shows the "Reader not connected" error in POS Totals screen in case the order draft is created but reader is not connected. 
2. Reader status is now observed, and the collect payment action is started automatically after the reader connects, allowing to tap the card to the reader.

Design specs: TfaZ4LUkEwEGrxfnEFzvJj-fi-2530_13218

⚠️ Don't merge until the base branch is merged into `trunk`.

### Steps to reproduce

1. Go to Woo POS (Menu more > Point of Sale)
2. Ensure reader is not connected
3. Add items to cart and click "Checkout"
4. After order draft is created, verify that "Reader not connected" error is displayed above totals table
5. Click "Connect to reader" button, and verify payment action starts automatically once the reader connects

### Testing information

1. "Reader not connected" error should be displayed in case order draft is created successfully, but reader is not connected.
2. In case order draft is not created successfully (e.g. try checking out in airplane mode), "Couldn't load totals" error is shown instead
3. "Connect to reader" button triggers reader connection flow
4. When reader connects, payment action should be started automatically, allowing to collect payment by tapping the card, and showing "Payment successful" screen

### The tests that have been performed

* Tested the "Reader not connected" shows up when reader is not connected
* Tested the "Reader not connected" shows up when reader suddenly disconnects (e.g. bluetooth gets turned off)
* Tested that "Connect to reader" action button works
* Tested the payment action starts automatically when reader connects

### Images/gif

|Light|Dark|
|---|---|
|Screenshot_20241203_164059|Screenshot_20241203_164044|

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">This PR is tagged with <code>status: do not merge</code> label(s).</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/common/composeui/WooPosTheme.kt [hunk 1]
```diff
@@ -186,6 +190,8 @@ private val DarkCustomColors = CustomColors(
     border = WooPosColors.oldGrayMedium,
     success = WooPosColors.greenNotFromPalette,
     error = WooPosColors.darkCustomColorsError,
+    totalsErrorBackground = WooPosColors.darkQuaternaryBackground,
+    totalsBackground = Color(0xFF1C1C1E),
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/common/composeui/WooPosTheme.kt:195]: ❓ Do we want to give this color a symantic name and use that name here?

**Review by AnirudhBhat — APPROVED** [review]
Thanks for working on this @samiuelson. Looks and tests well. I have just left one non-blocking comment.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/common/composeui/WooPosTheme.kt (hunk 1 above)
Comment (samiuelson) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/common/composeui/WooPosTheme.kt:195]: Fixed – ca316077eb553539fa3da5e97f3b8ed4072d4f65

**Pull request merged**